src/Reflector/Reflector.php

```diff
@@ -4,18 +4,131 @@
+
+        $classInfo = $this->sourceLocator->locateIdentifier($this, $identifier);
+        assert($classInfo instanceof ReflectionClass || $classInfo === null);
+
+        if ($classInfo === null) {
+            throw Exception\IdentifierNotFound::fromIdentifier($identifier);
+        }
+
+        return $classInfo;
+    }
+
+    /**
+     * Get all the classes available in the scope specified by the SourceLocator.
+     *
+     * @return ReflectionClass[]
+     */
+    public function reflectAllClasses(): array
+    {
+        /** @var ReflectionClass[] $allClasses */
+        $allClasses = $this->sourceLocator->locateIdentifiersByType(
+            $this,
+            new IdentifierType(IdentifierType::IDENTIFIER_CLASS),
+        );
+
+        return $allClasses;
+    }
+
     /**
-     * Create a reflection from the named identifier.
+     * Create a ReflectionFunction for the specified $functionName.
+     *
+     * @return ReflectionFunction
      *
      * @throws IdentifierNotFound
      */
-    public function reflect(string $identifierName): Reflection;
+    public function reflectFunction(string $identifierName): Reflection
+    {
+        $identifier = new Identifier($identifierName, new IdentifierType(IdentifierType::IDENTIFIER_FUNCTION));
+
+        $functionInfo = $this->sourceLocator->locateIdentifier($this, $identifier);
+        assert($functionInfo instanceof ReflectionFunction || $functionInfo === null);
+
+        if ($functionInfo === null) {
+            throw Exception\IdentifierNotFound::fromIdentifier($identifier);
+        }
+
+        return $functionInfo;
+    }
+
+    /**
+     * Get all the functions available in the scope specified by the SourceLocator.
+     *
+     * @return ReflectionFunction[]
+     */
+    public function reflectAllFunctions(): array
+    {
+        /** @var ReflectionFunction[] $allFunctions */
+        $allFunctions = $this->sourceLocator->locateIdentifiersByType(
+            $this,
+            new IdentifierType(IdentifierType::IDENTIFIER_FUNCTION),
+        );
+
+        return $allFunctions;
+    }
+
+    /**
+     * Create a ReflectionConstant for the specified $constantName.
+     *
+     * @return ReflectionConstant
+     *
+     * @throws IdentifierNotFound
+     */
+    public function reflectConstant(string $identifierName): Reflection
+    {
+        $identifier = new Identifier($identifierName, new IdentifierType(IdentifierType::IDENTIFIER_CONSTANT));
+
+        $constantInfo = $this->sourceLocator->locateIdentifier($this, $identifier);
+        assert($constantInfo instanceof ReflectionConstant || $constantInfo === null);
+
+        if ($constantInfo === null) {
+            throw Exception\IdentifierNotFound::fromIdentifier($identifier);
+        }
+
+        return $constantInfo;
+    }
+
+    /**
+     * Get all the constants available in the scope specified by the SourceLocator.
+     *
+     * @return array<int, ReflectionConstant>
```

Comment:
Since the signature changed, I think it might be a good idea to return an `iterable` here for future-proofing our ability to use a lazy iterator